modules/filesystem/bm.nu

```diff
@@ -24,28 +24,38 @@ export def list [] {
     open ($bm_path)
 }
 
+def os_home [] {
+  if ($nu.os-info.name == "windows" ) {
+    ($env.USERPROFILE)
+  } else {
+    ($env.HOME)
+  }
+}
+
 def get_path [] {
   $env.BM_PATH? |
   default (
     $env.XDG_DATA_HOME? |
     default (
-      $env.HOME? | path join ".local" "share" |
-      default (
-        $env.USERPROFILE? | path join "bm"
-      )
+      if $nu.os-info.name == windows {
+        ($env.USERPROFILE? | path join "bm")
+      } else {
+        ($env.HOME? | path join ".local" "share")
+      }
     )
   ) |
-    if (not ($in | path exists)) {
-      mkdir $in
-      $in
-    } | 
-    path join "bookmarks.nuon"
-  )
+  if (not ($in | path exists)) {
+    mkdir $in
+    $in
+  } else {
+    $in
+  }|
+  path join "bookmarks.nuon"
 }
 
 def save_path [] {
   $in |
-  update path { str replace $env.HOME '~' } |
+  update path { str replace os_home '~' } |
```

Comment:
I'm just guessing but I'm thinking it needs to be in parens to get it to evaluate.
```suggestion
  update path { str replace (os_home) '~' } |
```

Reply:
Nice catch!, updated the script.